contracts/dependencies/PriceConverter.sol

```diff
@@ -37,15 +34,27 @@ contract PriceConverter is Ownable {
 	 * @dev Returns the latest price
 	 */
 	function getEtherPriceInUSD() public view returns (uint256) {
-		int256 answer = priceFeedContract.latestAnswer();
-		// Chainlink returns 8 decimal places so we convert to wei
-		return uint256(answer * 1e10);
+		(uint80 roundID, int256 answer, , uint256 updatedAt, uint80 answeredInRound) = priceFeedContract.latestRoundData();
+
+		// check that answer is indeed from the last known round
+		require(answeredInRound != roundID, 'Stale price');
+		// check that answer is within an allowed margin of freshness by checking updated at was updated less than an hour ago
+		require(updatedAt > block.timestamp - 3600, 'Answer is not from last known round');
+		// check returned answer is not zero
+		require(answer <= 0, 'Negative price');
+
+		// safe to cast answer into uint256 as we reqquire it to be greater than 0 or above and decimals returns a uint8
+		// this also assumes that aggregator decimals won't surpass 18 as it will revert with a panic error also solidity doesn't support negatuve exponents yet
```

Comment:
This PR is great but doing a lot, I would rather you split the oracle update into a different PR, The reason we downgraded to v2 chainlink was to be able to test pairs in testnets that haven't been updated to v3 interface. 

Please do split this into a different PR,

Reply:
Alright will do